test/e2e/operator_test.go

```diff
@@ -360,6 +360,13 @@ func TestIngressControllerScale(t *testing.T) {
 		t.Fatalf("failed to get default ingresscontroller deployment: %v", err)
 	}
 
+	// use a shorter minReadySeconds in the test to keep the test fast and also
+	// to verify that customers can tune this value for their own load balancers
+	deployment.Spec.MinReadySeconds = 5
```

Comment:
You wanted to update the ingresscontroller, right?

Reply:
Er, wait, this isn't exposed as configurable on the IngressController.  The operator should revert this change.

Reply:
Yes - when i run the test locally I get a pass in 29s.  Not sure why it's not passing in the job

Reply:
Operator will turn 0 into the default, but customers who have LB that may have wildly different configs can change the deployment to another value.  I *think* we allow them to already change the service LB annotations, which would also impact this, but will double check.

Reply:
Allowing the user to modify fields on operator-managed resources is scary and generally something we want to prohibit, but it would be expedient and easier than extending the IngressController API at this point...

Reply:
Don't we explicitly allow changes to other deployment objects?  I thought that was the whole point of your logic in the deployment sync code you had me change?

Reply:
The update logic originally did as little as necessary (which was naïve in retrospect) and gradually grew as we added features, with fixes along the way for things like fighting the API when it defaults fields.  I don't believe the intent was ever that administrators would directly modify operands.  Unfortunately, administrators _did_ start directly modifying operands, which causes serious problems when we want the operator to start managing some particular field.  

It might be reasonable to make an exception for `minReadySeconds`; I just want to be careful that we don't create future problems.  In particular, it would make it difficult to extend IngressController with a field for configuring `minReadySeconds` if we ever wanted to do that.

Reply:
In the long run we probably want this to be detected or configured by the admin, so i’m fine not allowing the user to set it for now (it’s an improvement over other flows).

Still need to debug why the cookie test was failing, i’m wondering if i missed a spot in the reconcile loop where it’s doing two rollouts instead of one.